engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/types/TypeRepresentation.java

```diff
@@ -51,8 +51,7 @@ public QualifiedName getAssociatedType() {
    * <p>Instances that are assigned this type are built with one of the available constructors, but
    * statically we do not necessarily know which one.
    */
-  record AtomType(QualifiedName fqn, AtomTypeInterface typeInterface)
-      implements TypeRepresentation {
+  record AtomType(QualifiedName fqn) implements TypeRepresentation {
```

Comment:
Our type representation is no longer holding references to some complex `AtomTypeInterface` structure which internally was holding references to BindingsMap and was overall ugly.

Now the `TypeRepresentation` is a plain data type. The information about Atom type interface is instead encoded in the `StaticModuleScope`, like information about method already has been.